Model/Sales/MagentoOrderCollection.php

```diff
@@ -467,64 +350,19 @@ public function sendTrackEmails()
      * Update all the tracks that made created via the API
      *
      * @return $this
-     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Exception
      */
     public function updateMagentoTrack(): self
     {
-        foreach ($this->getShipmentsCollection() as $shipment) {
-            $trackCollection = $shipment->getAllTracks();
-            foreach ($trackCollection as $magentoTrack) {
-                $myParcelTrack = $this->myParcelCollection->getConsignmentByApiId(
-                    $magentoTrack->getData('myparcel_consignment_id')
-                );
-
-                if (! $myParcelTrack) {
-                    continue;
-                }
-
-                if ($myParcelTrack->getStatus()) {
-                    $magentoTrack->setData('myparcel_status', $myParcelTrack->getStatus());
-                }
-
-                if ($myParcelTrack->getBarcode()) {
-                    $magentoTrack->setTrackNumber($myParcelTrack->getBarcode());
-                }
-
-                $magentoTrack->save();
-            }
-        }
-
-        $this->updateGridByOrder();
-
-        return $this;
+        return $this->updateMagentoTrackByShipment($this->getShipmentsCollection());
```

Comment:
Zijn al deze public functies met `$this->doSomething($this->getSomething())` wel nodig? Worden ze vaker gebruikt? Anders kan je net zo goed alle `getSomething`s inlinen, dit lijkt een overbodige laag nu

Reply:
Volgens mij is het zo gekomen omdat de MagentoOrderCollection en MagentoShipmentCollection allebei de shipments op een andere manier verzamelen. Eerst was alle code ongeveer dubbel, die heb ik nu vooral naar MagentoCollection verplaatst, alleen die shipments worden dus verschillend verzameld en die stuur ik dan maar door als zodanig.